src/routing_table/mod.rs

```diff
@@ -968,78 +912,67 @@ impl RoutingTable {
     }
 
     fn split_our_section(&mut self, version: u64) -> Vec<XorName> {
-        if self.our_version != version {
+        if self.our_prefix.version() != version {
             debug!(
-                "{:?} Not splitting our section with {:?} ver. {}, \
+                "{:?} Not splitting our section with {:?}, \
                  update is for a different version: {}",
                 self.our_name,
                 self.our_prefix,
-                self.our_version,
                 version
             );
             return Vec::new(); // Wrong version.
         }
         let next_bit = self.our_name.bit(self.our_prefix.bit_count());
-        let other_prefix = self.our_prefix.pushed(!next_bit);
-        self.our_prefix = self.our_prefix.pushed(next_bit);
+        let other_prefix = self.our_prefix.pushed(!next_bit).with_incremented_version();
+        self.our_prefix = self.our_prefix.pushed(next_bit).with_incremented_version();
```

Comment:
will this cause our version increased twice?

Reply:
I don't think so. On line 926, the version of `self.our_prefix` is not modified, because `with_incremented_version` doesn't modify `self` in place.

Reply:
with_incremented_version will modify the prefix in place.
it's not modifying self just because pushed returned a copy.
It might be ok, but I feel it's bit confusing and unsafe (pushed has been called twice modifying our_prefix but version only increased by one).

Reply:
No, `with_incremented_version` does not modify the prefix in place. But anyhow, what would you suggest to be a less confusing way to do this?